app/routes/analysis/analysis.py

```diff
@@ -93,27 +102,46 @@ async def _zonal_statics(
         "end_date": end_date,
     }
 
-    response_payload_encoded = await _invoke_lambda(payload)
-    response_payload = json.loads(response_payload_encoded.decode())
+    try:
+        response = await _invoke_lambda(payload)
+    except httpx.TimeoutException:
+        raise HTTPException(500, "Query took too long to process.")
 
-    if response_payload["statusCode"] != 200:
+    resp_dict = response.json()
+
+    if response.status_code != 200 or resp_dict.get("statusCode") != 200:
```

Comment:
While it looks like Justin always returns a json like objection, there might be a case where the Lambda function fails and does not return a json like object but a string.

how about 
```python
try:
    response_data = response.json()["body"]["data"]
except KeyError, <HTTPX error for when response is not json-serializable>:
    logger.error(
           f"Raster analysis lambda experienced an error. Full response: {response.text}"
        )	        
       raise HTTPException(500, "Raster analysis geoprocessor experienced an error. See logs."	)

return Response(data=response_data)
```

Reply:
https://github.com/encode/httpx/blob/master/httpx/_models.py#L1110-L1118

They are just using json.loads so error would be `json.decoder.JSONDecodeError`